meta/schemas/device.yaml

```diff
@@ -0,0 +1,332 @@
+      - 57600/8N1
+      - 115200/8N1
+      - 1500000/8N1
+
+  sata_ports:
+    $id: '#/properties/sata_ports'
+    title: Sata Ports
+    type: integer
+    minimum: 0
+
+  gpio_pins:
+    $id: '#/properties/gpio_pins'
+    title: GPIO Pins
+    type: integer
+    minimum: 0
+
+  storage:
+    $id: '#/properties/storage'
+    title: Storage
+    type: array
+    items:
+      $ref: file:storage.yaml
+    minItems: 1
+
+  target:
+    $id: '#/properties/target'
+    type: array
+    items:
+      $ref: file:target.yaml
+    minItems: 1
+
+  vendor:
+    $id: '#/properties/vendor'
+    $ref: file:vendor.yaml
+
+  vlan:
+    $id: '#/properties/vlan'
+    title: VLAN
+    enum:
+      - True
+      - False
+      - 16
+      - 4096
+
+  audio_ports:
+    title: Audio Ports
+    type: array
+    items:
+      anyOf:
+        - type: object
+          required:
+            - type
+          properties:
+            type:
+              $id: '#/properties/audio_ports/items/anyOf/0/properties/type'
+              type: string
+              enum:
+                - HDMI
+                - Headphone
+                - I2S
+                - Line
+                - Line-In
+                - Line-Out
+                - Microphone
+                - S/PDIF
+                - other
+                - unknown
+            port_count:
+              $id: '#/properties/audio_ports/items/anyOf/0/properties/port_count'
+              title: port_count
+              type: integer
+              minimum: 1
+
+  video_ports:
+    title: Video Ports
+    type: array
+    items:
+      anyOf:
+        - type: object
+          required:
+            - type
+          properties:
+            type:
+              type: string
+              enum:
+                - Camera
+                - Composite
+                - DVI-D
+                - HDMI
+                - I2S
+                - LVDS
+                - MIPI
+                - DSI
+                - VGA
+                - other
+                - unknown
+            port_count:
+              title: port_count
+              type: integer
+              minimum: 1
+
+  phone_ports:
+    title: Phone Ports
+    type: array
+    items:
+      anyOf:
+        - type: object
+          required:
+            - type
+          properties:
+            type:
+              $id: '#/properties/phone_ports/items/anyOf/0/properties/type'
+              type: string
+              enum:
+                - DECT
+                - ISDN
+                - POTS
+                - other
+                - unknown
+            port_count:
+              $id: '#/properties/phone_ports/items/anyOf/0/properties/port_count'
+              title: port_count
+              type: integer
+              minimum: 1
+
+  bluetooth:
+    $id: '#/properties/bluetooth'
+    title: Bluetooth
+    type: string
+    enum:
+      - BLE
+      - "4.0"
+      - "4.1"
+      - "4.2"
+      - "5.0"
+      - "5.1"
+      - "5.2"
+
+  wifi:
```

Comment:
wifi, bluetooth, zigbee, lora, 802.11ah (HaLow), 802.11ad (MGWS), you-name-it are all wireless interfaces/networking standards and perhaps should be treated under one node?